### PR TITLE
Add fused_multi_transformer op to optimize transformer generation performance

### PR DESCRIPTION
**Pull request by wangxicoding**
### PR types

Performance optimization
### PR changes

OPs
### Describe

Add fused_multi_transformer to optimize ERNIE generation inference model performance.
API document:
火狐截图_2022-04-26T02-07-32 011Z
火狐截图_2022-04-26T02-08-34 694Z

### CODE REVIEW

File: paddle/fluid/operators/fused/fused_multi_transformer_op.cu [hunk 1]
```diff
@@ -0,0 +1,1327 @@
...
+      // step4. out_linear
+      out_linear_compute.ComputeForward(out_linear_weights[i], &fmha_out,
+                                        nullptr, buf1, nullptr);
+      AllReduce<T>(*buf1, ring_id, dev_ctx);
+#ifdef _DEBUG_FUSED_MULTI_TRANSFORMER
+      VLOG(0) << "step4";
+#endif
+
+      // step5. ln(residual + dropout(input + bias))
+      if (pre_layer_norm) {
```
Comment (ZzSean) [inline paddle/fluid/operators/fused/fused_multi_transformer_op.cu:1259]: 这种融合应该是pre_layer_norm为false的情况吧

**Review by ZzSean — APPROVED** [review]
LGTM

**Review by Aurelius84 — APPROVED** [review]
LGTM for data registeration

**Review by XieYunshen — APPROVED** [review]
LGTM for `set_tests_properties(test_static_model_parallel_fused_multi_transformer PROPERTIES TIMEOUT 120)`

**Review by jzhang533 — APPROVED** [review]
API is exposed in `paddle.incubate` namespace.
LGTM

**Review by XiaoguangHu01 — APPROVED** [review]
LGTM

**Pull request merged**